onnx/defs/tensor/defs.cc

```diff
@@ -471,55 +474,96 @@ Example 2:
 
 ONNX_OPERATOR_SET_SCHEMA(
     Slice,
-    1,
+    10,
     OpSchema()
-        .SetDoc(Slice_ver1_doc)
+        .SetDoc(Slice_ver10_doc)
         .Input(0, "data", "Tensor of data to extract slices from.", "T")
-        .Attr(
-            "axes",
-            "Axes that `starts` and `ends` apply to. "
-            "It's optional. If not present, will be treated as "
-            "[0, 1, ..., len(`starts`) - 1].",
-            AttributeProto::INTS,
-            OPTIONAL)
-        .Attr(
-            "starts",
-            "Starting indices of corresponding axis in `axes`",
-            AttributeProto::INTS)
-        .Attr(
-            "ends",
-            "Ending indices (exclusive) of corresponding axis in axes`",
-            AttributeProto::INTS)
+        .Input(1, "starts", "1-D tensor of starting indices of corresponding axis in `axes`", "Tind")
+        .Input(2, "ends", "1-D tensor of ending indices (exclusive) of corresponding axis in `axes`", "Tind")
+        .Input(3, "axes", "1-D tensor of axes that `starts` and `ends` apply to.", "Tind", OpSchema::Optional)
+        .Input(4, "steps", "1-D tensor of slice step of corresponding axis in `axes`. Default to 1. ", "Tind", OpSchema::Optional)
```

Comment:
so the "steps" is used to specify the axis order/step of slicing? if that's the case, can we just use/follow the "axes" indices order please?

Reply:
I'm not sure I understand the question correctly. Is your question suggesting that for a given axis `j = axes[i]`, `output[i]` should be computed based on `starts[j]`, `ends[j]`, `steps[j]`? If that's the case then yes that is the current design, I'll update the description to be more precise.